src/services/stringCompletions.ts

```diff
@@ -698,10 +720,17 @@ namespace ts.Completions.StringCompletions {
             getCompletionEntriesFromDirectories(root);
         }
 
-        // Also get all @types typings installed in visible node_modules directories
-        for (const packageJson of findPackageJsons(scriptPath, host)) {
-            const typesDir = combinePaths(getDirectoryPath(packageJson), "node_modules/@types");
-            getCompletionEntriesFromDirectories(typesDir);
+        if (require("module").findPnpApi?.(scriptPath)) {
+            for (const root of getPnpTypeRoots(scriptPath)) {
+                getCompletionEntriesFromDirectories(root);
+            }
+        }
+ else {

```

Comment:
This else seems massively distanced from it’s initiating block, with likelihood to degrade readability and maintainability.